llvm/lib/Target/AMDGPU/SIInstrInfo.cpp

```diff
@@ -3318,6 +3318,12 @@ bool SIInstrInfo::FoldImmediate(MachineInstr &UseMI, MachineInstr &DefMI,
       if (pseudoToMCOpcode(NewOpc) == -1)
         return false;
 
+      // V_FMAAK_F16_t16 takes VGPR_32_Lo128 operands, so the rewrite
+      // would also require restricting their register classes. For now
```

Comment:
So just handle this?

Reply:
I don't think the reasoning for real True16 is right, but the conclusion we don't want to shrink is right. VGPR_16_Lo128 (or something derived from it ) would still be used downstream. 
`dag InOperandList = (ins VSrcT_f16_Lo128_Deferred:$src0, VGPRSrc_16_Lo128:$src1, KImmFP16:$imm);`

We don't want to shrink pre-RA because it would restrict the register allocation to only use the bottom 128 vgprs

Reply:
This looks irrelevant to me. The comment aims to explain why we bail out for fake True16 and `VGPR_32_Lo128`, not `VGPR_16_Lo128`. The instruction in the condition is supposed to be changed to `V_FMAAK_F16_fake16` when we have one, and then removed along with the rest of the fake16 support.